src/main/java/RolloutInfo.java

```diff
@@ -0,0 +1,73 @@
+import java.io.StringReader;
+import javax.json.Json;
+import javax.json.JsonException;
+import javax.json.JsonObject;
+
+public class RolloutInfo {
+
+    public final GoogleURL googleurl;
+    public final OATHURL oathurl;
+    public final OTPKey otpkey;
+    public final String raw;
+    public String serial;
+    public String rolloutState;
+
+    public RolloutInfo(String json) {
+        this.raw = json;
+        this.googleurl = new GoogleURL();
+        this.oathurl = new OATHURL();
+        this.otpkey = new OTPKey();
+
+        if (json == null || json.isEmpty()) {
+            return;
+        }
+
+        JsonObject jsonObject;
+        try {
+            jsonObject = Json.createReader(new StringReader(json)).readObject();
+        } catch (JsonException | IllegalStateException e) {
+            e.printStackTrace();
```

Comment:
should we use the `logError()` function here as well?

Reply:
It cannot be accessed from there without a reference to the privacyIDEA object which would be unnecessary.
If the parsing fails, the object will contain the default values (empty strings) and the raw response which could still be inspected.